source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -781,11 +777,21 @@ ServerContextImpl::ServerContextImpl(Stats::Scope& scope,
                                      const std::vector<std::string>& server_names,
                                      TimeSource& time_source)
     : ContextImpl(scope, config, time_source), session_ticket_keys_(config.sessionTicketKeys()),
-      ocsp_staple_policy_(config.ocspStaplePolicy()) {
+      ocsp_staple_policy_(config.ocspStaplePolicy()), has_rsa_(false),
+      full_scan_certs_on_sni_mismatch_(config.fullScanCertsOnSNIMismatch()) {
   if (config.tlsCertificates().empty() && !config.capabilities().provides_certificates) {
     throw EnvoyException("Server TlsCertificates must have a certificate specified");
   }
 
+  for (auto& ctx : tls_contexts_) {
+    bssl::UniquePtr<EVP_PKEY> public_key(X509_get_pubkey(ctx.cert_chain_.get()));
```

Comment:
This crashes for cases when cert_chain is nullptr. It is missing nullptr check present in `ContextImpl::getCertChainInformation()`

Reply:
It already gets checked in loading phase. If the cert_chain is nullptr, envoy will log and  raise exception, you could check following code path https://github.com/envoyproxy/envoy/blob/84330823115f54dcfcfd7a2998ecad253d6431a1/source/extensions/transport_sockets/tls/context_impl.cc#L198-L204

So I don't think this will cause crash, and we can see similar code in constructor of its parent class, it checks nothing since     it get checked earlier.
https://github.com/envoyproxy/envoy/blob/84330823115f54dcfcfd7a2998ecad253d6431a1/source/extensions/transport_sockets/tls/context_impl.cc#L215-L216

I saw this change was reverted  by #24475 , but no enough context for me, could you elaborate the example config that it will crash?

cc @ggreenway @KBaichoo